### PR TITLE
dnfdaemon: Enhance comps.Group.list() method

### PR DESCRIPTION
**Pull request by m-blaha**
New accepted options:

- with_hidden - boolean, defaults to false
  whether include hidden (uservisible=false) groups into output
- scope - string, accepted values "all" (default), "installed",
"available"
- group_with_id - boolean, defaults to true
  whether patterns should be matched against group ids
- group_with_name - boolean, defaults to false
  whether patterns should be matched against group names
- contains_pkgs - list of strings
  only return groups containing given packages

Change in behavior:

- in case the scope is "all" (default), duplicated groups are removed
from the output. Duplicities basically mean that the group is
installed and previously both installed and available groups were
returned.

Also adds comps.Groups interface to the D-Bus API documentation and adjusts dnf5daemon-client accordingly.

Tests: https://github.com/rpm-software-management/dnf5/pull/1352

### CODE REVIEW

File: dnf5daemon-client/wrappers/dbus_group_wrapper.hpp [hunk 1]
```diff
@@ -51,7 +51,7 @@ class DbusGroupWrapper {
     std::string get_order() const { return rawdata.at("order"); }
     std::string get_langonly() const { return rawdata.at("langonly"); }
     // TODO(mblaha) proper installed value
-    bool get_installed() const { return false; }
+    bool get_installed() const { return rawdata.at("installed"); }
```
Comment (pkratoch) [inline dnf5daemon-client/wrappers/dbus_group_wrapper.hpp:54]: Can the TODO note be removed now?

File: dnf5daemon-client/commands/group/group_list.hpp [hunk 2]
```diff
@@ -22,20 +22,56 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
...
+};
+
+class GroupContainPkgsOption : public libdnf5::cli::session::AppendStringListOption {
+public:
+    explicit GroupContainPkgsOption(libdnf5::cli::session::Command & command)
+        : AppendStringListOption(
+              command,
+              "contains-pkgs",
+              '\0',
+              _("Show only groups containing packages with specified names. List option  , supports globs."),
```
Comment (pkratoch) [inline dnf5daemon-client/commands/group/group_list.hpp:59]: There are two extra spaces in the description.

File: dnf5daemon-server/dbus/interfaces/org.rpm.dnf.v0.comps.Group.xml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        Get list of groups that match to given filters.
+
+        Following options and filters are supported:
+
+            - attributes: list of strings
+                list of group attributes that are returned
+            - group_with_id: bool (default true)
+                match patterns against ids of the groups
+            - group_with_name: bool (default false)
+                match patterns against names of the groups
```
Comment (pkratoch) [inline dnf5daemon-server/dbus/interfaces/org.rpm.dnf.v0.comps.Group.xml:42]: I am a bit puzzled by the `group_with_id` and `group_with_name` option names, as I wouldn't intuitively know what they mean. The only thing similar I found were options for rpm filtering (`with_src`, `with_provides`, `with_filenames`...), but then there are similar ones for advisories that mean something different (`with_bz`, `with_cve`). Plus the `group_` prefix could be seen as "related to groups" or as a verb. What was the intention behind the naming?

**Comment by pkratoch** [comment]
I have just three small notes above, otherwise this looks good, thanks!

File: dnf5daemon-client/commands/group/group_list.hpp (hunk 2 above)
Comment (m-blaha) [reply dnf5daemon-client/commands/group/group_list.hpp:59]: Thanks.

File: dnf5daemon-client/wrappers/dbus_group_wrapper.hpp (hunk 1 above)
Comment (m-blaha) [reply dnf5daemon-client/wrappers/dbus_group_wrapper.hpp:54]: Sure, thanks.

File: dnf5daemon-server/dbus/interfaces/org.rpm.dnf.v0.comps.Group.xml (hunk 3 above)
Comment (m-blaha) [reply dnf5daemon-server/dbus/interfaces/org.rpm.dnf.v0.comps.Group.xml:42]: Thanks for the feedback!
The option names originate in `ResolveSpecSettings` which has the same attributes. But thinking about it again... you are right that the names in this context are puzzling.
How about renaming `group_with_id` -> `match_group_id`, `group_with_name`->`match_group_name` ? If you have another/better idea, it'd be appreciated.

Comment (pkratoch) [reply dnf5daemon-server/dbus/interfaces/org.rpm.dnf.v0.comps.Group.xml:42]: `match_group_id` and `match_group_name` sound great, I don't have anything better.

**Comment by m-blaha** [comment]
@pkratoch I've committed the changes you requested. Hopefully you don't mind rebase...

**Pull request merged**